### DISCUSSION TITLE
How to specify external port for KafkaContainer

### DISCUSSION THREAD

**Discussion by purplefox** [testcontainers/testcontainers-go#2399] [General]
Hi,

I am using KafkaContainer to start a Kafka broker and I access it via externalKafkaAddress:

```
kafkaContainer, err := kafka.RunContainer(ctx,
		kafka.WithClusterID("test-cluster"),
		testcontainers.WithImage("confluentinc/confluent-local:7.5.0"),
	)
brokers, err := kc.Brokers(context.Background())
require.NoError(t, err)
externalKafkaAddress := brokers[0]
```

This works ok, however I need external address to be a specific address that's the same each time, rather than a random unassigned port.

How can I configure KafkaContainer to expos the Kafka broker via a specific port?

Many thanks

**Comment by kiview** [comment]
Hi, you say:
> This works ok, however I need external address to be a specific address that's the same each time,

Can you elaborate on your use case, why that is necessary?
